pandas/core/groupby/generic.py

```diff
@@ -1585,6 +1585,9 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
             result = self._insert_inaxis_grouper(result)
             result.index = default_index(len(result))
 
+        if isinstance(self.obj, type(self.obj)):
+            result = self.obj._constructor(result)
```

Comment:
I'm -1 here. This makes an extra copy that is unnecessary if the user properly extends pandas. See https://github.com/pandas-dev/pandas/issues/59667#issuecomment-2324674201

Reply:
Agreed